lib/order.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'menu'
+
+class Order 
+
+  attr_reader :menu, :final_total
+
+  def initialize(menu = Menu.new)
+    @menu = menu.menu
+    @my_order = []
+    @total = []
+    @final_total = 0
+  end
+
+  def order_meal(*items)
+    items.each do |item|
+      if @menu.has_key?(item)
+        @my_order.push(item)
+      else
+        raise "that's not on the menu!"
+      end
+    end
+  end
+
+  def calculate_total
```

Comment:
Nice work!

You might be interested to know that `sum` can take a block so don't need to use `each`.

```ruby
@my_order.sum {| item| menu[item] }
```